### ISSUE TITLE
Implement `lu_factor` and `lu_solve` for Blockwise `Solve` rewrites

### ISSUE THREAD

**Issue by jessegrabowski** [pymc-devs/pytensor#582]
### Description

This blog post makes an important point about speed in cases where we want to repeatedly compute $A^{-1}b$ -- we should factorize the `A` matrix once, then recycle it for repeatedly solving. The specific factorization used in this case is LU factorization -- this can be verified by looking at the [function calls for dgesv](https://netlib.org/lapack/explore-html/d8/da6/group__gesv_ga831ce6a40e7fd16295752d18aed2d541.html) -- first dgetrf is called to perform an LU  factorization, then dtrsm to actually solve the system. In scipy, these correspond to `linalg.lu_factor` and `linalg.lu_solve`. [Here's a gist ](https://gist.github.com/jessegrabowski/2ed3b1fde4729394f2cb401f8a76bc21) showing a very simple speed test, suggesting 5-6x speedup by computing the LU decomposition once then recycling it many times for different $b$.

(Question: are other factorization schemes used in other solve cases (symmetric, positive-definite?)

All of this is relevant because of blockwise. This would be very low-hanging fruit for a rewrite in cases where a user has batch dimensions on the `b` matrix. 

In addition, gradient implementations exist for both [lu_factor ](https://github.com/pytorch/pytorch/blob/9eb842cbd6f80b3d8c3e88bf70e389465c86eb6f/torch/csrc/autograd/FunctionsManual.cpp#L6413
) and [lu_solve ](https://github.com/pytorch/pytorch/blob/9eb842cbd6f80b3d8c3e88bf70e389465c86eb6f/torch/csrc/autograd/FunctionsManual.cpp#L5658) in pytorch and jax ([lu_factor](https://jax.readthedocs.io/en/latest/_autosummary/jax.scipy.linalg.lu_factor.html), [lu_solve](https://jax.readthedocs.io/en/latest/_autosummary/jax.scipy.linalg.lu_solve.html). I haven't found an academic reference that shows where their equations come from, though.

**Comment by ricardoV94** [comment]
Re gradients, have to check whether the naive graph created by the solve is easy to parse/optimize, or if we would need to have eager gradients. 

Could be related to #559

**Comment by jessegrabowski** [comment]
@ricardoV94 we did this I think right?

**Comment by ricardoV94** [comment]
Yup

**Comment by ricardoV94** [comment]
Only took us 2 years
